src/modules/registrypreview/RegistryPreviewExt/dllmain.cpp

```diff
@@ -0,0 +1,253 @@
+// dllmain.cpp : Defines the entry point for the DLL application.
+#include "pch.h"
+#include <interface/powertoy_module_interface.h>
+#include <common/SettingsAPI/settings_objects.h>
+#include "trace.h"
+#include <common/interop/shared_constants.h>
+#include <common/utils/string_utils.h>
+#include <common/utils/winapi_error.h>
+#include <common/utils/logger_helper.h>
+#include <common/utils/EventWaiter.h>
+#include <common/utils/resources.h>
+
+#include "resource.h"
+#include "Constants.h"
+
+extern "C" IMAGE_DOS_HEADER __ImageBase;
+
+BOOL APIENTRY DllMain(HMODULE /*hModule*/, DWORD ul_reason_for_call, LPVOID /*lpReserved*/)
+{
+    switch (ul_reason_for_call)
+    {
+    case DLL_PROCESS_ATTACH:
+        Trace::RegisterProvider();
+        break;
+    case DLL_THREAD_ATTACH:
+    case DLL_THREAD_DETACH:
+        break;
+    case DLL_PROCESS_DETACH:
+        Trace::UnregisterProvider();
+        break;
+    }
+    return TRUE;
+}
+
+const static wchar_t* MODULE_NAME = L"RegistryPreview";
+const static wchar_t* MODULE_DESC = L"A quick little utility to visualize and edit complex Windows Registry files.";
+
+class RegistryPreviewModule : public PowertoyModuleIface
+{
+
+private:
+    bool m_enabled = false;
+
+    //Hotkey m_hotkey;
+    HANDLE m_hProcess;
+
+    HANDLE triggerEvent;
+    EventWaiter triggerEventWaiter;
+
+    bool is_process_running()
+    {
+        return WaitForSingleObject(m_hProcess, 0) == WAIT_TIMEOUT;
+    }
+
+    void launch_process()
```

Comment:
We need to check the policy here otherwise you can open it even if disabled.

Reply:
I'll fix this

Reply:
fixed